flink-table/flink-sql-gateway-api/src/main/java/org/apache/flink/table/gateway/api/results/serde/LogicalTypeJsonSerializer.java

```diff
@@ -0,0 +1,210 @@
+    public static final String FIELD_NAME_PRECISION = "precision";
+
+    // DECIMAL
+    public static final String FIELD_NAME_SCALE = "scale";
+
+    // MAP
+    public static final String FIELD_NAME_KEY_TYPE = "keyType";
+    public static final String FIELD_NAME_VALUE_TYPE = "valueType";
+
+    // ARRAY, MULTISET
+    public static final String FIELD_NAME_ELEMENT_TYPE = "elementType";
+
+    // ROW
+    public static final String FIELD_NAME_FIELDS = "fields";
+    public static final String FIELD_NAME_FIELD_NAME = "name";
+    public static final String FIELD_NAME_FIELD_TYPE = "fieldType";
+
+    // RAW
+    public static final String FIELD_NAME_CLASS = "class";
+    public static final String FIELD_NAME_SERIALIZER = "serializer";
+
+    public LogicalTypeJsonSerializer() {
+        super(LogicalType.class);
+    }
+
+    @Override
+    public void serialize(
+            LogicalType logicalType,
+            JsonGenerator jsonGenerator,
+            SerializerProvider serializerProvider)
+            throws IOException {
+        serializeInternal(logicalType, jsonGenerator);
+    }
+
+    private void serializeInternal(LogicalType logicalType, JsonGenerator jsonGenerator)
+            throws IOException {
+        jsonGenerator.writeStartObject();
+
+        // write common fields shared by all types
+        jsonGenerator.writeStringField(FIELD_NAME_TYPE_NAME, logicalType.getTypeRoot().name());
+        jsonGenerator.writeBooleanField(FIELD_NAME_NULLABLE, logicalType.isNullable());
+        // write special fields according to type root
+        switch (logicalType.getTypeRoot()) {
+            case BOOLEAN:
+            case TINYINT:
+            case SMALLINT:
+            case INTEGER:
+            case BIGINT:
+            case FLOAT:
+            case DOUBLE:
+            case DATE:
+            case NULL:
+                break;
+            case CHAR:
+                jsonGenerator.writeNumberField(
+                        FIELD_NAME_LENGTH, ((CharType) logicalType).getLength());
+                break;
+            case VARCHAR:
+                jsonGenerator.writeNumberField(
+                        FIELD_NAME_LENGTH, ((VarCharType) logicalType).getLength());
+                break;
+            case BINARY:
+                jsonGenerator.writeNumberField(
+                        FIELD_NAME_LENGTH, ((BinaryType) logicalType).getLength());
+                break;
+            case VARBINARY:
+                jsonGenerator.writeNumberField(
+                        FIELD_NAME_LENGTH, ((VarBinaryType) logicalType).getLength());
+                break;
+            case DECIMAL:
+                jsonGenerator.writeNumberField(
+                        FIELD_NAME_PRECISION, ((DecimalType) logicalType).getPrecision());
+                jsonGenerator.writeNumberField(
+                        FIELD_NAME_SCALE, ((DecimalType) logicalType).getScale());
+                break;
+            case TIMESTAMP_WITHOUT_TIME_ZONE:
```

Comment:
miss `TIME_WITHOUT_TIME_ZONE` and `TIMESTAMP_WITH_TIME_ZONE`

Reply:
Added.